libc/calls/getprogramexecutablename.greg.c

```diff
@@ -51,6 +52,12 @@ static inline int IsAlpha(int c) {
 }
 
 static inline void InitProgramExecutableNameImpl(void) {
+  if (__program_executable_name) {
+    /* already set by the loader */
+    return;
+  }
+  npassert(!issetugid());   // TODO(mrdomino): pathname security
```

Comment:
This could easily go after `IsWindows()` and `IsMetal()`... should it, or should this feature be the same on all platforms?

I know of no other problems with setuid/setgid programs, so this function seems like a fine place for the check.

I decided against printing a friendlier error message here for two reasons: one, this assert will hopefully be short-lived. Two, with assertions on, the assert failure should be clear enough already to someone who has wound up with a setuid/setgid ape program on their system.

Reply:
How is making the binary crash a better option? Can't we just use the exotic kernel guaranteed paths only when it's setuid?

Reply:
You mean actually implement the correct change rather than punt to later? Fine....

Reply:
Thank you!

Reply:
Ok, fixed.